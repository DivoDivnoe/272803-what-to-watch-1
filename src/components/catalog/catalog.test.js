import React from 'react';
import {create} from 'react-test-renderer';
import Catalog from './catalog.jsx';

const MockComponent = () => <div />;

const mock = {
  movies: [
    {
      name: `Fantastic Beasts`,
      posterImage: ``,
      genre: `Comedy`,
      previewVideoLink: ``,
    },
    {
      name: `Major Payne`,
      posterImage: ``,
      genre: `Comedy`,
      previewVideoLink: ``,
    },
  ],
  renderTabs: jest.fn(),
};

describe(`Catalog component`, () => {
  it(`renders correctly`, () => {
    const {movies, renderTabs} = mock;

    const tree = create(
        <Catalog
          movies={movies}
          renderTabs={renderTabs}
          renderButton={() => <MockComponent />}
        />,
        {createNodeMock: (el) => {
          return el;
        }}).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
