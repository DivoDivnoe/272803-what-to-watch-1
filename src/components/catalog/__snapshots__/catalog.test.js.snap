// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog component renders correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={null}
    >
      <video
        height="175"
        poster=""
        width="280"
      >
        Your browser does not support the video tag.
      </video>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Fantastic Beasts
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={null}
    >
      <video
        height="175"
        poster=""
        width="280"
      >
        Your browser does not support the video tag.
      </video>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Major Payne
        </a>
      </h3>
    </article>
  </div>
  <div />
</section>
`;
